### DISCUSSION TITLE
multi thread read-only access to a nlohmann::json object

### DISCUSSION THREAD

**Discussion by doj** [nlohmann/json#4095] [Q&A] [answered]
Hello JSON experts,

I would like to create a nlohmann::json object, by parsing a file.
Then I would like to access that object from multiple threads simultaneously, but only with a read-only access of looking up data.
Think about loading a configuration file once at process start time, then using it with multiple threads to look at configured values.

Is that supported? I think that the nlohmann::json implementation typically used STL containers to store information, and these are typically safe to use with read-only access with multiple threads. At least on the Linux gcc compiler I am using (I don't need to care about other platforms).

**Comment by nlohmann** [comment, marked as answer]
Yes, just treat it as any other STL container. Reading from multiple threads is safe.
